openlibrary/plugins/openlibrary/home.py

```diff
@@ -140,7 +140,7 @@ def loans_carousel(loans=None, css_id="CarouselLoans"):
     """Generates 'Your Loans' carousel on home page"""
     _loans = loans or []
     books = [format_book_data(web.ctx.site.get(loan['book']))
```

Comment:
I think a loop is the right way to go. I debated and stuck with the list comprehension, but I agree that it's really not equipped to handle the needs in an elegant and DRY way.